version: "3.8"

services:
  app:
    build: ./.docker/app
    image: my_python/app:1.0
    container_name: "my_python_app"
    ports:
      - "8080:80"
      - "8000:8000"
    stdin_open: true
    tty: true
    working_dir: "/workspace"
    volumes:
      - ./:/workspace
    depends_on:
      - db

  db:
    build: ./.docker/db
    image: my_python/mysql:1.0
    container_name: 'my_python_db'
    expose:
      - "3306"
    ports:
      - 3306:3306
    environment:
      - MYSQL_ROOT_PASSWORD=root
      - MYSQL_DATABASE=test
      - MYSQL_USER=test
      - MYSQL_PASSWORD=test
      - TZ="Asia/Tokyo"
    volumes:
      - ./.docker/db/my.cnf:/etc/mysql/conf.d/my.cnf
      - ./.docker/db/database:/var/lib/mysql
      - ./.docker/db/init:/docker-entrypoint-initdb.d

  phpmyadmin:
    container_name: 'my_python_phpmyadmin'
    image: phpmyadmin:latest
    environment:
      - PMA_ARBITRARY=1
      - PMA_HOST=db
      - PMA_USER=root
      - PMA_PASSWORD=root
    links:
      - db
    ports:
      - 8081:80

  mailhog:
    container_name: 'my_python_mailhog'
    image: mailhog/mailhog
    ports:
      - "8025:8025"
